File: JavaAndScala/src/main/scala/org/lang/scala/conclusion/design/factory/abstracts/MacFactory.scala
package org.lang.scala.conclusion.design.factory.abstracts

import org.lang.scala.conclusion.design.factory.abstracts.common.MacBorder
import org.lang.scala.conclusion.design.factory.abstracts.common.MacButton

/**
 * 	This is a stand-alone class to implement abstract factory pattern
 *  -- This class must implement a parent trait
 *  
 * 	@author VinceYuan
 */
class MacFactory extends BrandFactory {
  	
	override def getButton() = new MacButton()
	override def getBorder() = new MacBorder()
}

File: JavaAndScala/src/main/java/com/java/se/conclusion/design/factory/abstracts/MacFactory.java
package com.java.se.conclusion.design.factory.abstracts;

import com.java.se.conclusion.design.factory.abstracts.common.Border;
import com.java.se.conclusion.design.factory.abstracts.common.Button;
import com.java.se.conclusion.design.factory.abstracts.common.MacBorder;
import com.java.se.conclusion.design.factory.abstracts.common.MacButton;

/**
 * 	This is a class to implement abstract factory pattern
 *  -- This class must implement a parent interface
 *  
 * @author VinceYuan
 *
 */
public class MacFactory implements BrandFactory {

	@Override
	public Button getButton() {
		return new MacButton();
	}

	@Override
	public Border getBorder() {
		return new MacBorder();
	}
}
